pkg/api/apiv1/cancellation.go

```diff
@@ -0,0 +1,82 @@
+package apiv1
+
+import (
+	"crypto/rand"
+	"encoding/json"
+	"net/http"
+	"time"
+
+	"github.com/inngest/inngest/pkg/cqrs"
+	"github.com/inngest/inngest/pkg/publicerr"
+	"github.com/oklog/ulid/v2"
+)
+
+func (a api) GetCancellations(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	auth, err := a.opts.AuthFinder(ctx)
+	if err != nil {
+		_ = publicerr.WriteHTTP(w, publicerr.Wrap(err, 401, "No auth found"))
+		return
+	}
+
+	all, err := a.opts.CancellationReadWriter.Cancellations(ctx, auth.WorkspaceID())
+	if err != nil {
+		_ = publicerr.WriteHTTP(w, publicerr.Wrap(err, 500, "Error listing cancellations"))
+		return
+	}
+
+	_ = json.NewEncoder(w).Encode(all)
+
+}
+
+type CreateCancellationBody struct {
+	// AppID is the client ID specified via the SDK in the app that defines the function.
+	AppID string `json:"appID"`
+	// FunctionID is the function ID string specified in configuration via the SDK.
+	FunctionID string     `json:"functionID"`
+	From       *time.Time `json:"from"`
+	To         time.Time  `json:"to"`
+	If         *string    `json:"if,omitempty"`
+}
+
+func (a api) CreateCancellation(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	auth, err := a.opts.AuthFinder(ctx)
+	if err != nil {
+		_ = publicerr.WriteHTTP(w, publicerr.Wrap(err, 401, "No auth found"))
+		return
+	}
+
+	opts := CreateCancellationBody{}
+	if err := json.NewDecoder(r.Body).Decode(&opts); err != nil {
+		_ = publicerr.WriteHTTP(w, publicerr.Wrap(err, 400, "Invalid cancellation request"))
+		return
+	}
+
+	fn, err := a.opts.FunctionReader.GetFunctionByExternalID(
+		ctx,
+		auth.WorkspaceID(),
+		opts.AppID,
+		opts.FunctionID,
+	)
+	if err != nil {
+		_ = publicerr.WriteHTTP(w, publicerr.Wrap(err, 404, "Function not found"))
+		return
+	}
+
+	// Create a new cancellation for the given function ID
+	cancel := cqrs.Cancellation{
+		ID:          ulid.MustNew(ulid.Now(), rand.Reader),
+		WorkspaceID: auth.WorkspaceID(),
+		FunctionID:  fn.ID,
```

Comment:
**issue (dx)**: I believe this is the function's internal `uuid`, which is confusing to return as the payload accepts the same named param (`function_id`), but it's a slug there and a uuid here. We should make this the slug in the response. Ideally, it would also contain `app_id`

Reply:
Changed, thanks.  We need new names for referring to internal vs external IDs.